setools/policyrep/terule.pxi

```diff
@@ -536,8 +544,15 @@ cdef class FileNameTERule(BaseTERule):
             yield self
 
     def statement(self):
-        return "{0.ruletype} {0.source} {0.target}:{0.tclass} {0.default} {0.filename};". \
-            format(self)
+        name_match_str = "???"
+        if self.name_match == sepol.NAME_TRANS_MATCH_EXACT:
+            name_match_str = "";
+        if self.name_match == sepol.NAME_TRANS_MATCH_PREFIX:
+            name_match_str = " PREFIX";
+        elif self.name_match == sepol.NAME_TRANS_MATCH_SUFFIX:
+            name_match_str = " SUFFIX";
+        return "{0.ruletype} {0.source} {0.target}:{0.tclass} {0.default} {0.filename}{1};". \
+            format(self, name_match_str)
```

Comment:
This is how the policy rule looks like? It may just be a failure of searching, but I can't find an example of the syntax for this (in .te not .cil).

Reply:
Yes, it is a new syntax as prefix/suffix rules have not been supported before. Examples of the new syntax can be seen in [7th patch][1] of the series I linked. It is not merged yet, due to waiting for kernel to be merged first.

[1]: https://lore.kernel.org/selinux/c3fab035f9d1b18c55b5d691fa72aef9f468377a.1687251081.git.juraj@jurajmarcin.com/